Library/Keychain.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+import Security
+
+public enum KeychainError: Error {
+  case unexpectedStatus(OSStatus)
+  case unableToDecodePasswordData
+
+  public var errorMessage: String {
+    switch self {
+    case let .unexpectedStatus(status):
+      let errString = SecCopyErrorMessageString(status, nil) as String?
+      return errString ?? "Unknown keychain error"
+    case .unableToDecodePasswordData:
+      return "Unable to decode password data from keychain."
+    }
+  }
+}
+
+public struct Keychain {
+  private static var serviceName: String {
+    return Bundle.main.bundleIdentifier ?? "com.kickstarter.kickstarter"
```

Comment:
The documentation for this is poor; I think it can be an arbitrary string, but the bundle ID seems like a nice value to use.